src/json/from_json.cr

```diff
@@ -244,13 +244,48 @@ def NamedTuple.new(pull : JSON::PullParser)
 end
 
 def Enum.new(pull : JSON::PullParser)
-  case pull.kind
-  when .int?
-    from_value(pull.read_int)
-  when .string?
-    parse(pull.read_string)
-  else
-    raise "Expecting int or string in JSON for #{self.class}, not #{pull.kind}"
+  {% if @type.annotation(Flags) %}
+    if pull.kind.begin_array?
+      value = {{ @type }}::None
+      pull.read_array do
+        unless pull.kind.string?
+          pull.raise "Expected String, not #{pull.kind}"
```

Comment:
Doesn't read_string do this check already?

Reply:
Indeed, the JSON implementation can be simplified in several places.